src/front/wgsl.rs

```diff
@@ -354,6 +355,38 @@ pub enum Scope {
     GeneralExpr,
 }
 
+fn check_constant_types<'a>(
+    inner: &crate::ConstantInner,
+    type_inner: &crate::TypeInner
+) -> Result<(), Error<'a>> {
+    match (inner, type_inner) {
```

Comment:
another way to write this could be:
```rust
let expected_kind = match inner {
  crate::ConstantInner::Sint(_) => crate::ScalarKind::Sint,
  ...
  crate::ConstantInner::Composite(inner) => {
     return Something;
  }
};
match type_inner {
  crate::TypeInner::Scalar {  kind, width: _ } if kind == expected_kind => Ok(()),
  _=> Err(...),
}
```

It's up to you to pick a preference here.

Reply:
I'd like to leave it as-is for now, since I'm guessing that the code will need to change substantially to incorporate type checking for composite types anyways.